### PR TITLE
Bootstrapping Compliance Info 

### PR DESCRIPTION
**Pull request by v-zhgup**
## Description
*  Includes changes for fetching compliance info from back end for a subscription 
* Added alias to GAI parameters
* Bug fix for GAI -infoType AttestationInfo
## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK/AzSKInfo/AzSKInfo.ps1 [hunk 1]
```diff
@@ -182,9 +185,22 @@ function Get-AzSKInfo
 							$Full = $false
 						}
 						$complianceInfo = [ComplianceInfo]::new($SubscriptionId, $PSCmdlet.MyInvocation, $Full);
-						if ($complianceInfo) 
+						if($PSCmdlet.MyInvocation.BoundParameters["Sync"] -and $PSCmdlet.MyInvocation.BoundParameters["Sync"].IsPresent)
```
Comment (VishalHaibatpure) [inline src/AzSK/AzSKInfo/AzSKInfo.ps1]: u can directly use $Sync here

Comment (v-zhgup) [reply src/AzSK/AzSKInfo/AzSKInfo.ps1]: Done

File: src/AzSK/Framework/Core/AzSKInfo/ComplianceInfo.ps1 [hunk 2]
```diff
@@ -326,6 +326,22 @@ class ComplianceInfo: CommandBase
 		$ComplianceMessage.ComplianceCount = $ComplianceCount
 		$this.ComplianceMessageSummary += $ComplianceMessage
 	}
+	hidden [void] UpdateStorageComplianceData([string] $subId, [InvocationInfo] $invocationContext)
```
Comment (VishalHaibatpure) [inline src/AzSK/Framework/Core/AzSKInfo/ComplianceInfo.ps1]: SubId and Context must be taken from base

File: src/AzSK/Framework/Helpers/ComplianceReportHelper.ps1 [hunk 3]
```diff
@@ -371,4 +386,87 @@ class ComplianceReportHelper: ComplianceBase
 			return "";
 		}	
 	}
+
+	hidden [void] FetchComplianceStateFromDb([string] $subId, $invocationContext)
+	{
+		
+		$result =  [RemoteAPIHelper]::GetComplianceSnapshot($subId)
```
Comment (VishalHaibatpure) [inline src/AzSK/Framework/Helpers/ComplianceReportHelper.ps1]: Apply check on result

File: src/AzSK/Framework/Helpers/ComplianceReportHelper.ps1 [hunk 4]
```diff
@@ -371,4 +386,87 @@ class ComplianceReportHelper: ComplianceBase
...
+		$Complianceinfo = ConvertFrom-Json -InputObject $result
+		$ComplianceState = New-Object -TypeName "System.Collections.Generic.List[SVTEventContext]";
+		$subContext= [SubscriptionContext]:: new();
```
Comment (VishalHaibatpure) [inline src/AzSK/Framework/Helpers/ComplianceReportHelper.ps1]: $this.SubscriptionContext

File: src/AzSK/Framework/Core/AzSKInfo/ComplianceInfo.ps1 (hunk 2 above)
Comment (v-zhgup) [reply src/AzSK/Framework/Core/AzSKInfo/ComplianceInfo.ps1]: Done

File: src/AzSK/Framework/Helpers/ComplianceReportHelper.ps1 (hunk 3 above)
Comment (v-zhgup) [reply src/AzSK/Framework/Helpers/ComplianceReportHelper.ps1]: Done.

File: src/AzSK/Framework/Helpers/ComplianceReportHelper.ps1 (hunk 4 above)
Comment (v-zhgup) [reply src/AzSK/Framework/Helpers/ComplianceReportHelper.ps1]: done

File: src/AzSK/Framework/Core/AzSKInfo/ComplianceInfo.ps1 [hunk 5]
```diff
@@ -10,12 +10,15 @@ class ComplianceInfo: CommandBase
 	hidden $baselineControls = @();
 	hidden [PSObject] $ControlSettings
 	hidden [PSObject] $EmptyResource = @();
+	hidden [InvocationInfo] $InvocationContext
```
Comment (VishalHaibatpure) [inline src/AzSK/Framework/Core/AzSKInfo/ComplianceInfo.ps1]: Take it from base classes

File: src/AzSK/Framework/Helpers/ComplianceReportHelper.ps1 [hunk 6]
```diff
@@ -371,4 +386,100 @@ class ComplianceReportHelper: ComplianceBase
 			return "";
 		}	
 	}
+
+	hidden [string] FetchComplianceStateFromDb()
+	{
+		
+			$result =  [RemoteAPIHelper]::GetComplianceSnapshot($this.SubscriptionContext.SubscriptionId)
+			if(-not [string]::IsNullOrEmpty($result) -and $result -ne '[]')
```
Comment (VishalHaibatpure) [inline src/AzSK/Framework/Helpers/ComplianceReportHelper.ps1]: Change result object type

File: src/AzSK/Framework/Helpers/ComplianceReportHelper.ps1 [hunk 7]
```diff
@@ -371,4 +386,100 @@ class ComplianceReportHelper: ComplianceBase
...
+						{
+							$CResult.IsControlInGrace = $false
+						}
+						$StateData.AttestedBy = $item.attestedBy;
+						$StateData.AttestedDate = $item.attestedDate
+						$StateData.Justification = $item.Justification
+						$CResult.StateManagement.AttestedStateData=$StateData
+						$controlDetails.ControlId=$item.ControlId
+						$CResult.CurrentSessionContext.IsLatestPSModule =$true
+						$CResult.CurrentSessionContext.Permissions.HasRequiredAccess = $true
```
Comment (VishalHaibatpure) [inline src/AzSK/Framework/Helpers/ComplianceReportHelper.ps1]: DB scan version with PSGallery version check

File: src/AzSK/Framework/Helpers/ComplianceReportHelper.ps1 [hunk 8]
```diff
@@ -371,4 +386,100 @@ class ComplianceReportHelper: ComplianceBase
...
+						$CResult.CurrentSessionContext.Permissions.HasAttestationWritePermissions = $this.azskStorageInstance.HaveWritePermissions
+						$CResult.CurrentSessionContext.Permissions.HasAttestationReadPermissions = $this.azskStorageInstance.HaveWritePermissions 
+						$controlDetails.Id=$item.ControlIntId
+						$controlDetails.ControlSeverity=$item.ControlSeverity
+						$SVTEvent.ControlResults = $CResult;
+						if($item.IsBaselineControl -eq 1)
```
Comment (VishalHaibatpure) [inline src/AzSK/Framework/Helpers/ComplianceReportHelper.ps1]: Direct assignment

File: src/AzSK/Framework/Helpers/RemoteApiHelper.ps1 [hunk 9]
```diff
@@ -9,26 +9,47 @@ class RemoteApiHelper {
...
+                $result = Invoke-WebRequest -Uri $([RemoteApiHelper]::ApiBaseEndpoint + $uri) `
+                    -Method Post `
+                    -Body $content `
+                    -ContentType $type `
+                    -Headers @{"Authorization" = "Bearer $accessToken"} `
+                    -UseBasicParsing
+                return $result
+            }
+            catch {
+                return "ERROR"
```
Comment (VishalHaibatpure) [inline src/AzSK/Framework/Helpers/RemoteApiHelper.ps1]: Add task to print error in log file

**Pull request merged**